### PR TITLE
Complete Construction of Courseflow Frontend Webpage

### PR DESCRIPTION
**Pull request by queser9**
_Any italic text should be deleted from the final Pull Request text, including this line_

# Description

Complete Construction of Courseflow Frontend Webpage，once the interfaces are added by the team member the webpage can be viewed from the On Track main interface.

Fixes # (issue)

## Type of change

_Please delete options that are not relevant._

- [ ] New feature (non-breaking change which adds functionality)
# How Has This Been Tested?

Import and Declare Component: In the doubtfire-angular.module.ts file, import and declare the courseflow component to ensure it is correctly registered in the module.

Add Route Configuration: In the doubtfire-angular.module.ts file, add route configuration to allow access to the courseflow component via a specific URL. This includes setting up a new route path, such as /test-courseflow, and associating it with the courseflow component.

Start Development Server: Use the Angular CLI command ng serve to start the development server, enabling local testing.

Access Test Page: Open a web browser and navigate to http://localhost:4200/test-courseflow to view the rendered courseflow component and verify its display and functionality without dependency on interfaces.

## Testing Checklist:

- [ ] Tested in latest Safari

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have requested a review from @macite and @jakerenzella on the Pull Request

### CODE REVIEW

**Comment by ZHONGYU1111** [comment]
The new component appears to meet the requirements by adding a tab to display elective units. The interface and structure seem appropriate for the functionality needed.The code is well-organized with separate components for each part of the UI, which aligns with good Angular practices
Component Structure: The new elective-units component is well-structured. It follows a similar pattern to the existing core-units component, ensuring consistency in the side panel design.
CSS Styling: The styling for the new component is clean and consistent with other parts of the application. The use of CSS classes for margins and padding maintains uniformity.
Suggestions for Improvement
Type Safety: Ensure that all TypeScript types are defined explicitly, particularly for inputs and outputs of components. This will enhance maintainability and reduce potential bugs.
Usability: The layout is clean, making it super easy for users to switch between core and elective units.
Suggestions
In summary, your code essentially meets the requirements of the task.

**Comment by maddernd** [comment]
This needs screenshots showing what it looks like and where it will be installed, also have we written any tests for these new components? How is it interacting with the existing services and models to store and get data?

**Comment by JoshuaBourquin** [comment]
I’m not exactly sure what changes you have made within this PR. It looks as if you have attempted to create many components within one PR for the courseflow frontend but it appears that most components are unfinished.

An example of this is in **trimester-container-component.ts**:
```ts
import { Component } from '@angular/core';

@Component({
  selector: 'app-trimester-container',
  templateUrl: './trimester-container.component.html',
  styleUrls: ['./trimester-container.component.css']
})
export class TrimesterContainerComponent {
  // 这里可以添加组件相关的逻辑
}
```

Also included in this example, there is a non-English comment, which is not useful for any other developers. Please ensure that all comments you make are in English.

There is also some dummy data included within the **courseflow.component.ts** file, which should be removed:
```ts
export class CourseflowComponent {
  courseName = "Sample Course Name";
  creditPointsAchieved = 5;
  totalCreditPoints = 24;
  coreUnits: Unit[] = [
    { name: "Core Unit 1", draggable: true },
    { name: "Core Unit 2", draggable: true },
    { name: "Core Unit 3", draggable: true }
  ];
  electiveUnits: Unit[] = [
    { name: "Elective Unit 1", draggable: true },
    { name: "Elective Unit 2", draggable: true },
    { name: "Elective Unit 3", draggable: true }
  ];
  years: Year[] = [
    { name: "Year 1", trimesters: [{ name: "Trimester 1", units: [] }, { name: "Trimester 2", units: [] }] },
    { name: "Year 2", trimesters: [{ name: "Trimester 1", units: [] }, { name: "Trimester 2", units: [] }] }
  ];
```
This data should come from the retrieving values from the backend, not created within typescript.

Within the testing part of your PR, you have stated the route to access the courseflow page is **/test-courseflow**. When submitting an item under a PR, it should be in it's final form and not in a testing form. This route should be changed to **/courseflow**.

Overall, for these components to be adequately reviewed they will have to be submitted under individual PR's and not as a group. Currently it is very difficult to identify what each component is doing in their incomplete state. There is much more work to be done before this PR can be accepted.
